app/models/mixins/authentication_mixin.rb

```diff
@@ -12,6 +12,33 @@ def self.authentication_check_schedule
       assocs = zone.respond_to?(assoc) ? zone.send(assoc) : []
       assocs.each { |a| a.authentication_check_types_queue(:attempt => 1) }
     end
+
+    def self.queue_authentication_check(args, user_id, zone)
```

Comment:
@jntullo 
This is making the call synchronously, its going to block the UI till the Authentication is done. Should we be using a notification approach as opposed to blocking the UI.

Also there is a put_authentication_check that also queues a credential validation. That gets dispatched from a scheduler. https://github.com/ManageIQ/manageiq/blob/master/app/models/mixins/authentication_mixin.rb#L245

For the name queue_authentication_check if we are going to block with an MiqTask could we call it validate_credentials_task to keep it distinct from the Queue.